src/transformers/trainer.py

```diff
@@ -421,6 +433,14 @@ def __init__(
         # force device and distributed setup init explicitly
         args._setup_devices
 
+        if tokenizer is not None:
+            if processing_class is not None:
+                raise ValueError(
+                    "You cannot specify both `tokenizer` and `processing_class` at the same time. Please use `processing_class`."
+                )
+            logger.warning("`tokenizer` is now deprecated, please use `processing_class` instead.")
```

Comment:
do we want futur warning?

Reply:
Maybe say that it will be removed in the v5 ?

Reply:
I'll move it to a future warning and specify v5 removal :)